### ISSUE TITLE
[a11y]: Contained List [all variants] - semantics not announced by the VO in iPhone

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13938]
### Package

@carbon/react

### Browser

iPhone / Safari

### Operating System

MacOS

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology

Voice Over

### Description

VO iPhone in Safari does not identify the semantics of the UL present in the page. These are announced only as text without any semantics. The same is working in JAWS and in android talkback Samsung phone with chrome. Note: this issue was verified in android to cross check the screen reader behavior.

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### Reproduction/example

Record `Number 26` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

User Impact: when the semantic structure of the element is not conveyed programmatically, Blind screen reader user would not be able to perceive it

Generally VO is expected to announce the beginning and the end of the list when we navigate using Left/right swipe. Note: It seems to be working in Unordered list component. And in the following example link as well.
https://codepen.io/svinkle/pen/YxJEdY

Press ESCAPE to close this message. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
